crates/sui-network/src/state_sync/mod.rs

```diff
@@ -1228,14 +1229,22 @@ async fn sync_checkpoint_contents<S>(
                             info!("unable to sync contents of checkpoint through state sync {}", checkpoint.sequence_number());
 
                         }
-                        // Retry contents sync on failure.
-                        checkpoint_contents_tasks.push_front(sync_one_checkpoint_contents(
-                            network.clone(),
-                            &store,
-                            peer_heights.clone(),
-                            timeout,
-                            checkpoint,
-                        ));
+
+                        if store.get_highest_synced_checkpoint()
```

Comment:
rather than adding extra special-case logic to the control flow when the checkpoint contents tasks returns an error, would it work to leave this code the same and make the checkpoint contents function itself load and return an already-synced checkpoint from the store? the latter to me seems more intuitive and less "magical" (i.e. you get to avoid the complexity of treating an error as not-acutally-an-error)

Reply:
I pushed a commit for this